pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/TaskGeneratorUtils.java

```diff
@@ -68,4 +73,30 @@
     }
     return runningSegments;
   }
+
+  public static Map<String, List<String>> getScheduledSegmentsMap(@Nonnull String taskType,
```

Comment:
Return `Map<String, Set<String>>` for fast lookup in the generator?

Reply:
fixed